lib/gui/components/safe-webview/safe-webview.js

```diff
@@ -54,63 +65,75 @@ class SafeWebview extends react.PureComponent {
    * @returns {react.Element}
    */
   render() {
-    if (this.state.shouldLoad) {
-      const url = new URL(this.props.src);
-
-      // We set the 'etcher-version' GET parameter here.
-      url.searchParams.set('etcher-version', packageJSON.version);
-
-      return react.createElement('webview', {
-        ref: 'webview',
-        src: url.href
-      }, []);
-    }
-
-    // We have to return null explicitly, undefined is an error in React.
-    return null;
+    return react.createElement('webview', {
+      ref: 'webview',
+      style: {
+        flex: this.state.shouldShow ? null : '0 1',
+        width: this.state.shouldShow ? null : '0',
+        height: this.state.shouldShow ? null : '0'
+      }
+    }, []);
   }
 
   /**
-   * @summary Add the Webview events if there is an element
+   * @summary Add the Webview events
    */
   componentDidMount() {
 
-    // There is no element to add events to if 'shouldLoad' is false.
-    if (this.state.shouldLoad) {
+    // Events React is unaware of have to be handled manually
+    this.refs.webview.addEventListener('did-fail-load', this.didFailLoad);
+    this.refs.webview.addEventListener('did-get-response-details', this.didGetResponseDetails);
+    this.refs.webview.addEventListener('new-window', this.constructor.newWindow);
+    this.refs.webview.addEventListener('console-message', this.constructor.consoleMessage);
 
-      // Events React is unaware of have to be handled manually
-      this.refs.webview.addEventListener('did-fail-load', this.didFailLoad);
-      this.refs.webview.addEventListener('did-get-response-details', this.didGetResponseDetails);
-      this.refs.webview.addEventListener('new-window', this.constructor.newWindow);
-      this.refs.webview.addEventListener('console-message', this.constructor.consoleMessage);
+    // Use the 'success-banner' session
+    this.refs.webview.partition = 'persist:success-banner';
 
-    }
+    // It's important that this comes after the partition setting, otherwise it will
+    // use another session and we can't change it without destroying the element again
+    this.refs.webview.src = this.makeURL();
   }
 
   /**
-   * @summary Remove the Webview events if there is an element
+   * @summary Refresh the webview if we are navigating away from the success page
+   * @param {Object} nextProps - upcoming properties
    */
-  componentWillUnmount() {
+  componentWillReceiveProps(nextProps) {
+    if (this.props.currentStateName === 'success' && nextProps.currentStateName !== 'success') {
```

Comment:
This component is hard-coding a state name that is set outside of the component (in the Angular.js app), which doesn't look right. It looks like we should make the component take a boolean attribute to determine when it should be refreshed, and move the fact that in our use case that depends on the `success` state outside of this component.